chapter4.adoc

```diff
@@ -123,107 +145,71 @@ domain
-[width="100%",cols="25%,75%",options="header",]
-|===
-|*`MTTL2` Entry Type* |*Description, `INFO` and `TYPE` field encoding*
-|`1G_allow` a|
-_The 1G range of address is allowed for the domain._
-The `INFO` field must be 0.
-When configuring 1G ranges, RDSM ensures that 16 `MTTL2` entries, each
-corresponding to 64M of address space, have identical `TYPE` field values.
+6. The _mpte_ is a valid leaf _mpte_. Fetch the access-permissions for the
+physical address per the steps described below:
 
-|`1G_disallow` a|
-_The 1G range of address is not allowed for the domain._
-The `INFO` field must be 0.
-When configuring 1G ranges, RDSM ensures that 16 `MTTL2` entries, each
-corresponding to 64M of address space, have identical `TYPE` field values.
+* if _i_=1, and the _mpte.type_ field directly specifies the access-permissions
+for 1GB page regions (via 32 MTTL2 entries with identical _mpte.type_ values -
+see <<Smmtt-rw-l2-encoding>>); go to step 7, else
 
-|`MTT_L1_DIR` a|
-The `INFO` field provides the PPN of the `MTTL1` page.
-Entries of the `MTTL1` page hold a 2-bit `PERM` field to indicate the access
-for the supervisor domain (described in the `MTTL1` entry <<mtt-l1>>).
+* if _i_=1, and the _mpte.type_ field value is `2M_PAGES`, then the
+_mpte.info_[31:0] field contains 16 entries of 2-bit access-permissions encoding
+for 16 2M address regions (see <<Smmtt-rw-l2-encoding>>); go to step 7, else
 
-|`2M_PAGES` a|
-_The 64M range of address space is partitioned into 2M pages where each
-page has access allowed/not._
-The `INFO` field bits 31:0 holds 1 bit per 2M address range to indicate
-access disallowed(0b) or allowed (1b). `INFO` field bits 43:32 are
-reserved (must be zero).
-|===
+* If _i_=0, the _mpte_ contains a 4-bit entry that holds the access-permission
+encodings for 4KB pages. The 4-bit access-permission encoding for the _pa_ is
+held in _mpte_ and indexed via _pa.pn[i]_. The encodings are specified in
+<<Smmtt-rw-l1-encoding>> table.
 
-`MTTL1` table is populated if 4KiB page confidential pages are required
-for the supervisor domain.
+7. Determine if the requested physical memory access is allowed per the
+access-permissions. If access is not permitted, stop and raise a page-fault
+exception corresponding to the original access type.
 
-`MTTL1` entry is indexed using PA[25:12] and each entry is a 2-bit `PERM` field
-in the referenced page. The bits 25:16 are used to select a 64-bit field
-in the page and bits 16:12 are used to select a 2 bit `PERM` field in those 64
-bits. Thus, there are 2 `PERM` bits for each 4 KiB page. The encoding is as
-follows:
+8. The access is allowed per the `MTT` lookup.
 
-* 00b - the 4K page specifies access is *not allowed* for the domain
-* 01b - the 4K page specifies access is *allowed* for the domain
-* 1xb - *reserved* (access causes access violation).
+All implicit accesses to the non-leaf memory tracking table data structures in
+this algorithm are performed using width NON_LEAF_PTE_SIZE.
```

Comment:
This may be problematic for RV32.

Reply:
will fix by making MTT L2, L1 entries XLEN dependent